tests/drivers/gpio/gpio_more_loops/src/main.c

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(gpio_loops, LOG_LEVEL_INF);
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/ztest.h>
+
+#if !DT_NODE_EXISTS(DT_NODELABEL(out_gpios)) || !DT_NODE_EXISTS(DT_NODELABEL(in_gpios))
+#error "Unsupported board: test_gpio node is not defined"
+#endif
+
+static const struct gpio_dt_spec out1_spec = GPIO_DT_SPEC_GET_BY_IDX(DT_NODELABEL(out_gpios),
+											gpios, 0);
+static const struct gpio_dt_spec in1_spec = GPIO_DT_SPEC_GET_BY_IDX(DT_NODELABEL(in_gpios),
+											gpios, 0);
+
+#if (CONFIG_NUMBER_OF_LOOPBACKS >= 2)
+#define LOOP_2_EXISTS
+static const struct gpio_dt_spec out2_spec = GPIO_DT_SPEC_GET_BY_IDX(DT_NODELABEL(out_gpios),
+											gpios, 1);
+static const struct gpio_dt_spec in2_spec = GPIO_DT_SPEC_GET_BY_IDX(DT_NODELABEL(in_gpios),
+											gpios, 1);
+#endif
```

Comment:
You could greatly simplify this by using `DT_FOREACH_PROP_ELEM()` to fill an array of something like:
```
const struct gpio_dt_spec out_pins[] = {
        DT_FOREACH_PROP_ELEM_SEP(DT_PATH(zephyr_user), out_gpios, GPIO_DT_SPEC_GET_BY_IDX, (,))
};
const struct gpio_dt_spec in_pins[] = {
        DT_FOREACH_PROP_ELEM_SEP(DT_PATH(zephyr_user), in_gpios, GPIO_DT_SPEC_GET_BY_IDX, (,))
};
BUILD_ASSERT(ARRAY_SIZE(in_pins) == ARRAY_SIZE(out_pins), "mismatched in/out pairs");
const uint8_t npairs = RRAY_SIZE(in_pins)M
```

Reply:
Done